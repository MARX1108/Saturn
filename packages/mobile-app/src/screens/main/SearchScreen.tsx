import { Ionicons } from '@expo/vector-icons';
import StyledText from '../../components/ui/StyledText';
import { useTheme } from '../../theme/ThemeContext';

const SearchScreen: React.FC = () => {
  const theme = useTheme();
  // ... existing state and handlers ...

  // Render empty state
  const renderEmptyComponent = () => {
    if (isLoading) return null;

    return (
      <View style={styles.emptyContainer}>
        <Ionicons
          name="search-outline"
          size={48}
          color={theme.colors.textSecondary}
          style={styles.emptyIcon}
        />
        <StyledText
          weight="medium"
          color={theme.colors.textSecondary}
          style={styles.emptyText}
        >
          No results found
        </StyledText>
        <StyledText
          color={theme.colors.textSecondary}
          style={styles.emptySubtext}
        >
          Try searching with different keywords
        </StyledText>
      </View>
    );
  };

  // ... rest of the component ...

  return (
    <SafeAreaView
      style={[styles.container, { backgroundColor: theme.colors.background }]}
      edges={['top']}
    >
      {/* ... existing search input ... */}
      <FlatList
        data={searchResults}
        renderItem={renderUserItem}
        keyExtractor={item => item.id}
        contentContainerStyle={styles.listContainer}
        ListEmptyComponent={renderEmptyComponent}
        refreshControl={
          <RefreshControl
            refreshing={isRefreshing}
            onRefresh={handleRefresh}
            colors={[theme.colors.primary]}
            tintColor={theme.colors.primary}
          />
        }
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  // ... existing styles ...
  emptyContainer: {
    padding: 32,
    alignItems: 'center',
    justifyContent: 'center',
  },
  emptyIcon: {
    marginBottom: 16,
  },
  emptyText: {
    fontSize: 18,
    marginBottom: 8,
  },
  emptySubtext: {
    fontSize: 14,
    textAlign: 'center',
  },
  // ... rest of the styles ...
});
