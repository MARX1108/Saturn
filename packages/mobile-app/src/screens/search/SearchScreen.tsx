import React from 'react';
import { View, StyleSheet, Platform } from 'react-native';
import { useTheme } from '../../theme/ThemeContext';
import ScreenWrapper from '../../components/ui/ScreenWrapper';
import TextInputWrapper from '../../components/ui/TextInputWrapper';
import { Ionicons } from '@expo/vector-icons';

const SearchScreen: React.FC = () => {
  const theme = useTheme();

  return (
    <ScreenWrapper style={styles.screen}>
      <View style={styles.headerContainer}>
        <View style={styles.searchContainer}>
          <Ionicons
            name="search"
            size={20}
            color={theme.colors.textSecondary}
            style={styles.searchIcon}
          />
          <TextInputWrapper
            placeholder="Search users..."
            value=""
            onChangeText={() => {}}
            autoCapitalize="none"
            autoCorrect={false}
            returnKeyType="search"
            style={[styles.searchInput, { color: theme.colors.text }]}
            containerStyle={[
              styles.searchInputContainer,
              { backgroundColor: theme.colors.surfaceVariant },
            ]}
          />
        </View>
      </View>
      <View
        style={[
          styles.resultsContainer,
          { backgroundColor: theme.colors.background },
        ]}
      />
    </ScreenWrapper>
  );
};

const styles = StyleSheet.create({
  screen: {
    flex: 1,
  },
  headerContainer: {
    paddingHorizontal: 16,
    paddingTop: Platform.OS === 'ios' ? 16 : 8,
    paddingBottom: 8,
  },
  searchContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  searchIcon: {
    position: 'absolute',
    left: 16,
    zIndex: 1,
  },
  searchInputContainer: {
    marginBottom: 0,
    borderRadius: 12,
    borderWidth: 0,
  },
  searchInput: {
    paddingLeft: 40,
    height: 40,
    fontSize: 16,
  },
  resultsContainer: {
    flex: 1,
  },
});

export default SearchScreen;
