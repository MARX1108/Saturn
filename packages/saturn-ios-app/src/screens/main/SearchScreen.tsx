import React, { useState } from 'react';
import {
  SafeAreaView,
  StyleSheet,
  Text,
  View,
  TextInput,
  Platform,
} from 'react-native';

export default function SearchScreen(): React.JSX.Element {
  const [searchQuery, setSearchQuery] = useState('');

  const handleSearchChange = (query: string) => {
    setSearchQuery(query);
    if (query.length > 0) {
      console.log('Search query:', query);
    }
  };

  return (
    <SafeAreaView style={styles.container} testID="search-screen-container">
      <Text style={styles.headerText} testID="search-screen-header">
        User Search
      </Text>
      <View style={styles.inputContainer} testID="search-input-container">
        <TextInput
          style={styles.textInput}
          placeholder="Search users by username or display name..."
          placeholderTextColor="#666666"
          value={searchQuery}
          onChangeText={handleSearchChange}
          autoCapitalize="none"
          autoCorrect={false}
          returnKeyType="search"
          testID="search-input-field"
        />
      </View>
      <View style={styles.resultsContainer} testID="search-results-container">
        {searchQuery.length > 0 ? (
          <Text
            style={styles.placeholderText}
            testID="search-results-placeholder"
          >
            {`Searching for "${searchQuery}"...`}
          </Text>
        ) : (
          <Text
            style={styles.placeholderText}
            testID="search-results-placeholder"
          >
            Enter a query to search for users.
          </Text>
        )}
        <View style={styles.debugContainer} testID="search-debug-container">
          <Text testID="search-debug-text">
            Debug info: SearchScreen is rendering
          </Text>
        </View>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f5f5f5',
  },
  headerText: {
    fontSize: 24,
    color: '#121212',
    fontWeight: 'bold',
    textAlign: 'center',
    marginVertical: 16,
  },
  inputContainer: {
    padding: 16,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: '#e0e0e0',
  },
  textInput: {
    fontSize: 16,
    padding: Platform.OS === 'ios' ? 12 : 8,
    backgroundColor: '#FFFFFF',
    borderRadius: 8,
    color: '#121212',
    borderWidth: 1,
    borderColor: '#e0e0e0',
  },
  resultsContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 16,
  },
  placeholderText: {
    fontSize: 16,
    color: '#666666',
    textAlign: 'center',
    marginBottom: 16,
  },
  debugContainer: {
    padding: 10,
    backgroundColor: '#f0f0f0',
    borderRadius: 8,
  },
});
