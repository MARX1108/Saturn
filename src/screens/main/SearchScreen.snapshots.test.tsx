import React from 'react';
import { render, act } from '@testing-library/react-native';
import SearchScreen from './SearchScreen';

// Mock the navigation
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: jest.fn(),
    }),
  };
});

describe('SearchScreen Snapshots', () => {
  it('renders correctly with empty search', () => {
    const { toJSON } = render(<SearchScreen />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with search text', () => {
    const { toJSON, getByTestId } = render(<SearchScreen />);

    // Use act to properly handle state updates
    act(() => {
      const searchInput = getByTestId('search-input-field');
      searchInput.props.onChangeText('test query');
    });

    expect(toJSON()).toMatchSnapshot();
  });
});
