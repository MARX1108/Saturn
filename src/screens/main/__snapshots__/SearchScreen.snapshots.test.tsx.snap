// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchScreen Snapshots renders correctly with empty search 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
    }
  }
  testID="search-screen-container"
>
  <Text
    style={
      {
        "color": "#121212",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginVertical": 16,
        "textAlign": "center",
      }
    }
    testID="search-screen-header"
  >
    User Search
  </Text>
  <View
    style={
      {
        "borderBottomColor": "#e0e0e0",
        "borderBottomWidth": 0.5,
        "padding": 16,
      }
    }
    testID="search-input-container"
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Search users by username or display name..."
      placeholderTextColor="#666666"
      returnKeyType="search"
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#e0e0e0",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "#121212",
          "fontSize": 16,
          "padding": 12,
        }
      }
      testID="search-input-field"
      value=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      }
    }
    testID="search-results-container"
  >
    <Text
      style={
        {
          "color": "#666666",
          "fontSize": 16,
          "marginBottom": 16,
          "textAlign": "center",
        }
      }
      testID="search-results-placeholder"
    >
      Enter a query to search for users.
    </Text>
    <View
      style={
        {
          "backgroundColor": "#f0f0f0",
          "borderRadius": 8,
          "padding": 10,
        }
      }
      testID="search-debug-container"
    >
      <Text
        testID="search-debug-text"
      >
        Debug info: SearchScreen is rendering
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`SearchScreen Snapshots renders correctly with search text 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
    }
  }
  testID="search-screen-container"
>
  <Text
    style={
      {
        "color": "#121212",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginVertical": 16,
        "textAlign": "center",
      }
    }
    testID="search-screen-header"
  >
    User Search
  </Text>
  <View
    style={
      {
        "borderBottomColor": "#e0e0e0",
        "borderBottomWidth": 0.5,
        "padding": 16,
      }
    }
    testID="search-input-container"
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Search users by username or display name..."
      placeholderTextColor="#666666"
      returnKeyType="search"
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#e0e0e0",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "#121212",
          "fontSize": 16,
          "padding": 12,
        }
      }
      testID="search-input-field"
      value="test query"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      }
    }
    testID="search-results-container"
  >
    <Text
      style={
        {
          "color": "#666666",
          "fontSize": 16,
          "marginBottom": 16,
          "textAlign": "center",
        }
      }
      testID="search-results-placeholder"
    >
      Searching for "
      test query
      "...
    </Text>
    <View
      style={
        {
          "backgroundColor": "#f0f0f0",
          "borderRadius": 8,
          "padding": 10,
        }
      }
      testID="search-debug-container"
    >
      <Text
        testID="search-debug-text"
      >
        Debug info: SearchScreen is rendering
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
